compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeHashVisitor.java

```diff
@@ -412,7 +412,7 @@ public Integer visit(BTupleType type) {
             return 0;
         }
         List<Integer> tupleTypesHashes = getOrderedTypesHashes(type.getTupleTypes());
-        Integer hash = hash(baseHash(type), tupleTypesHashes, visit(type.restType), type.flags);
+        Integer hash = hash(baseHash(type), tupleTypesHashes, visit(type.restType), type.flags, type.tsymbol);
```

Comment:
This adds the address of the `type.tsymbol` to the hash right. I believe the idea of the hash visitor is to have a hash which does not change between compilations. You may want to add `type.tsymbol.name.getValue()` like value?

Reply:
I believe there were cases where the `tsymbol` could be `null` here so we couldn't always get the name

Reply:
If it is null hash value can be zero for the name.

Reply:
Let's continue the discussion in the original PR with this change
- https://github.com/ballerina-platform/ballerina-lang/pull/37041